app/controllers/application_controller.rb

```diff
@@ -3,6 +3,8 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
 
+  before_action :do_not_cache
```

Comment:
It doesn't apply to assets: those aren't served by the controllers, so even if we do serve them from the app (which I'm not sure is the case in production) they get the correct headers.